### PR TITLE
beater: always use modelindexer for es output

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Enable the new Elasticsearch output by default, removing configuration to switch between this and libbeat. We only use libbeat outputs for other output types.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
~- [ ] Documentation has been updated~ Opened https://github.com/elastic/apm-server/issues/6662

## How to test these changes

N/A. Covered by integration tests, system tests.

## Related issues

#6360

### CODE REVIEW

**Review by marclop — APPROVED** [review]
Looks good, very exciting to see this turned on by default 🙂

File: cmd/root.go [hunk 1]
```diff
@@ -58,12 +56,13 @@ var libbeatConfigOverrides = func() []cfgfile.ConditionalOverride {
...
-				return true
+				return os.Getenv(cloudEnv) != ""
 			},
 			Config: func() *common.Config {
-				m := map[string]interface{}{}
-				cloudValues(m)
-				return common.MustNewConfigFrom(m)
+				return common.MustNewConfigFrom(map[string]interface{}{
+					// default to medium compression on cloud
+					"output.elasticsearch.compression_level": 5,
```
[the remark this answers is not in the document]
Comment (simitt) [reply cmd/root.go:64]: ++
We discussed a while back that we generally would want to align cloud and on-prem default values where possible. Any reason we don't also want to default to medium compression for other deployments?

[the remark this answers is not in the document]
Comment (axw) [reply cmd/root.go:64]: I don't think there's a good reason not to. I was going to do it in a follow up so we could observe the impact on benchmarks, if any, separately.

[the remark this answers is not in the document]
Comment (simitt) [reply cmd/root.go:64]: sounds good

**Review by simitt — APPROVED** [review]

**Pull request merged**